DG update - Chi Sern

ngchisern opened the pull request:
* Update logic component diagrams
* Update acknowledgement 
* Update model component's description
* Add a feature description in the implementation section
* Delete unnecessary tutorials in gettingstart page

ngchisern commented:
1. Model Class Diagram:
![ModelClassDiagram](https://user-images.githubusercontent.com/64767959/140467574-1e9543ed-a1c0-4cc4-8f9a-c9957600b036.png)

2. DeleteSequenceDiagram:
![DeleteSequenceDiagram](https://user-images.githubusercontent.com/64767959/140467041-d09f9d6d-430e-4c18-97e0-af2beeb904f8.png)

3. DeleteTaskSequenceDiagram:
![DeleteTaskSequenceDiagram](https://user-images.githubusercontent.com/64767959/140467081-54988a3b-7d8f-4393-9470-26ce9479e212.png)

4. EditSequenceDiagram:
![EditSequenceDiagram](https://user-images.githubusercontent.com/64767959/140467101-f9225330-3cfe-410e-939b-db30a069903f.png)

5. Person Class Diagram:
![PersonClassDiagram](https://user-images.githubusercontent.com/64767959/140467131-9bd35690-dad0-495b-8b9f-f298d6b1d5f2.png)

6. Task Class Diagram:
![TaskClassDiagram](https://user-images.githubusercontent.com/64767959/140467157-d0e2865c-aa03-421d-a6b4-da75ec340920.png)

7. Order Class Diagram:
![OrderClassDiagram](https://user-images.githubusercontent.com/64767959/140467401-048c140a-c9f1-4885-90f0-ee7a2c76d71c.png)

Tanishq4331 commented on docs/DeveloperGuide.md:
```diff
@@ -119,11 +120,11 @@ How the `Logic` component works:
+The Sequence Diagram below illustrates the interactions within the `Logic` component for the `execute("deletetask 1")` API call.
 
-![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteSequenceDiagram.png)
+![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteTaskSequenceDiagram.png)
```
Does the caption also need to be updated? 

```suggestion
![Interactions Inside the Logic Component for the `deletetask 1` Command](images/DeleteTaskSequenceDiagram.png)
```

ngchisern replied:
You're right

Tanishq4331 commented on docs/DeveloperGuide.md:
```diff
@@ -137,26 +138,30 @@ How the parsing works:
+
+* stores these three types of data in SalesNote
+  * Address book data i.e., all `Person` objects (which are contained in a `UniquePersonList` object).
+  * Task book data i.e., all `Tasks` objects (which are contained in a `UniqueTaskList` object).
```
Using `Task` and `Order` in this section instead of `Tasks` and `Orders` might be more consistent

Tanishq4331 commented on docs/DeveloperGuide.md:
```diff
@@ -236,116 +241,50 @@ is below:
+### Update Person Changes in Order List and Task List
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+SalesNote's clients are directly referenced in orders. Any changes in the clients through user commands should be propagated through the Order list.
```
```suggestion
SalesNote's clients are directly referenced in orders. Any changes in the clients through user commands should be propagated to the Order list.
```

Tanishq4331 commented on docs/DeveloperGuide.md:
```diff
@@ -236,116 +241,50 @@ is below:
+SalesNote's clients are directly referenced in orders. Any changes in the clients through user commands should be propagated through the Order list.
 
-Step 1. The user launches the application for the first time. The `VersionedSalesNote` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+* When a client is deleted, his/her orders and the tasks linked to the orders will be deleted as well.
```
Just a note - using "their" instead of "his/her" might be simpler.  But this is not really important so feel free to ignore.

Tanishq4331 reviewed (changes requested):
Great Work! Just a few notes: 

- If possible, it might be good to fix the overlap between *all and *filtered near the Person class in the Model Diagram. 
- The client commands also need to be updated to the new explicit command format
- I'm not too sure about this, but shouldn't there be solid diamonds for all the Task and Order fields (for composition)?

ngchisern commented:
made changes in line with Tanishq's suggestionss

ngchisern commented:
@Tanishq4331 I don't think there should be a composition relation between `task`/`order` with all their fields. 
e.g., When we delete the `order` class, we shouldn't delete the `label` class straightaway because the `task` class might still need it. 
Let me know if I am wrong!

ngchisern commented:
Updated Model Class Diagram: 
![ModelClassDiagram](https://user-images.githubusercontent.com/64767959/140504100-c15fc554-e1aa-4973-9bbc-00769bdafe52.png)

Updated DeleteSequenceDiagram:
![DeleteSequenceDiagram](https://user-images.githubusercontent.com/64767959/140504153-c9dabaa0-4da5-4ad9-9e5d-5a07f0b94505.png)

Updated DeleteTaskSequenceDiagram:
![DeleteTaskSequenceDiagram](https://user-images.githubusercontent.com/64767959/140504204-8fe58ecb-9e63-4979-94d4-a8b2c13463d3.png)

Updated EditSequenceDiagram:
![EditSequenceDiagram](https://user-images.githubusercontent.com/64767959/140504224-7ef32286-99b1-421b-b1d7-2ba4653a56d7.png)

Tanishq4331 commented:
Correct me if I misunderstood your example, but wouldn't the related tasks themselves be deleted when the order is deleted - which means the `label` object associated with it is no longer required and can also be deleted? 

As for the other fields, I was thinking the `order` being composed of a Date, Id, and IsComplete is conceptually similar to a `Person` being composed of all its fields. Same for the `task`.  Are these relationships different from those in the `Person` class?

ngchisern commented:
We are talking about the class diagram, so we shouldn't talk about 'object' in this context.
I believe that the difference is that the fields of the `Person` class are **only** referenced by the `Person` class itself while let's say, the `label` class is referenced by both `Task` and `Order` classes. 

If we happen to want to remove the `Order` class from our codebase, we shouldn't delete the `Label` class because the `Task` class might still need it. So I guess there is no composition relationship here.

Tanishq4331 commented:
Going through the notes, it seems like composition refers to object deletion instead of class deletion.

![image](https://user-images.githubusercontent.com/62829987/140511156-17fdf062-9ed8-41b3-bbdc-270ffc77b0b8.png)

ngchisern commented:
Oh no, seems like I've misunderstood the concept. Thanks for clarifying that for me!

ngchisern commented:
I've updated the diagrams. Thanks again.
![OrderClassDiagram](https://user-images.githubusercontent.com/64767959/140512199-81e5ad40-2254-4851-88c9-c28a84f4be2c.png)
![TaskClassDiagram](https://user-images.githubusercontent.com/64767959/140512208-caae982b-d7ac-4d37-b278-434ae9822614.png)

Tanishq4331 reviewed (approved):
Thanks for the changes! LGTM
